telegram/ext/_dispatcher.py

```diff
@@ -297,6 +324,18 @@ def get_instance(cls) -> 'Dispatcher':
             return cls.__singleton()  # type: ignore[return-value] # pylint: disable=not-callable
         raise RuntimeError(f'{cls.__name__} not initialized or multiple instances exist')
 
+    def _set_chat_data(self, data: DefaultDict[int, CD]) -> None:
+        """Used for assigning a new value to the underlying chat_data dictionary. Also updates
+        the read-only mapping."""
+        self._chat_data = data
+        self.chat_data = MappingProxyType(self._chat_data)
```

Comment:
This is basically what a setter for a `_chat_data` property would have to look like. though from https://github.com/python-telegram-bot/python-telegram-bot/pull/2852#discussion_r785294597 I rather had the impression that you wanted to go with `update` anyway.
looking at the current implementation, I'm even more in favor of that, as it would shorten the implementation & make it more readable …

Reply:
> This is basically what a setter for a _chat_data property would have to look like

yeah but the actual property would be something like:
``` python
@property
def _chat_data(self):
    if self.persistence:
        return self.persistence.get_chat_data()
    return defaultdict(self.context_types.chat_data)
```
which is again rebuilding a new object everytime

> looking at the current implementation, I'm even more in favor of that, as it would shorten the implementation & make it more readable …

on top of this, we can still do 3. from https://github.com/python-telegram-bot/python-telegram-bot/pull/2852#discussion_r785294597 I guess

Reply:
mh, no, I rather had in mind something like

```python
@property
def _chat_data(self):
    return self.__chat_data

@_chat_data.setter
def __chat_data(self, value):
    self.__chat_data = value
    self.chat_data = MappingProxyType(self.__chat_data)
```

but yeah, then let's just go for 3 :)